pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -1157,12 +1157,19 @@ func LoadIPSecKeys(log *slog.Logger, r io.Reader) (int, uint8, error) {
 		ipSecKey.Spi = spi
 		ipSecKey.ESN = esn
 
-		if ipSecKeysGlobal[""] != nil {
-			oldSpi = ipSecKeysGlobal[""].Spi
+		if oldKey, ok := ipSecKeysGlobal[""]; ok {
+			if reflect.DeepEqual(oldKey, ipSecKey) {
+				continue
+			}
+
+			if oldKey.Spi == spi {
+				return 0, 0, fmt.Errorf("invalid SPI: new value must differ from the previous one")
+			}
+
+			ipSecKeysRemovalTime[oldKey.Spi] = time.Now()
 		}
-		ipSecKeysGlobal[""] = ipSecKey
 
-		ipSecKeysRemovalTime[oldSpi] = time.Now()
```

Comment:
Moving this line inside the if may change the behavior since it will now be skipped in case the key didn't change. Is there any wider impact?

Reply:
I made a different PR discussing it https://github.com/cilium/cilium/pull/34652.
At this point, here we introduce just the check for the same SPI, for which we want to return an error. In this case, I would not insert the `oldKey.Spi`  in the `ipSecKeysRemovalTime` map, since it is not actually removed.
What's your opinion on this?

Reply:
Yep, it looks good now.